test/e2e/incluster/rbac_test.go

```diff
@@ -0,0 +1,185 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build e2e
+// +build e2e
+
+package incluster
+
+import (
+	"context"
+	"crypto/tls"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	contourv1 "github.com/projectcontour/contour/apis/projectcontour/v1"
+	contourv1alpha1 "github.com/projectcontour/contour/apis/projectcontour/v1alpha1"
+	"github.com/projectcontour/contour/test/e2e"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
```

Comment:
for reviewers: this was a cheap way to do this via behavioral testing, but maybe an even cheaper way would be to assert on the content of the rbac rules applied?

still could add gateway to this, but wanted to wait until i heard people's thoughts on this method

Reply:
I like the style you have here since it's less likely to end up as just a copy-paste of whatever the defined RBAC is.